app/controllers/user_sessions_controller.rb

```diff
@@ -50,7 +50,13 @@ def deliver_access_details
   end
 
   def user_session
-    @user_session ||= UserSession.new(params[:user_session])
+    @user_session ||= UserSession.new(user_session_params.to_unsafe_hash)
```

Comment:
Not as bad is it sounds - it is converting a set of parameters that have been permitted - to a hash.  The word unsafe is misleading